src/System.Windows.Extensions/src/System/Security/Cryptography/X509Certificates/X509Certificate2UI.cs

```diff
@@ -0,0 +1,154 @@
+            DisplayX509Certificate(X509Utils.GetCertContextSafeHandle(certificate), IntPtr.Zero);
+        }
+
+        public static void DisplayCertificate(X509Certificate2 certificate, IntPtr hwndParent)
+        {
+            if (certificate == null)
+                throw new ArgumentNullException(nameof(certificate));
+            DisplayX509Certificate(X509Utils.GetCertContextSafeHandle(certificate), hwndParent);
+        }
+
+        public static X509Certificate2Collection SelectFromCollection(X509Certificate2Collection certificates, string title, string message, X509SelectionFlag selectionFlag)
+        {
+            return SelectFromCollectionHelper(certificates, title, message, selectionFlag, IntPtr.Zero);
+        }
+
+        public static X509Certificate2Collection SelectFromCollection(X509Certificate2Collection certificates, string title, string message, X509SelectionFlag selectionFlag, IntPtr hwndParent)
+        {
+            return SelectFromCollectionHelper(certificates, title, message, selectionFlag, hwndParent);
+        }
+
+        private static void DisplayX509Certificate(SafeCertContextHandle safeCertContext, IntPtr hwndParent)
+        {
+            if (safeCertContext.IsInvalid)
+                throw new CryptographicException(SR.Format(SR.Cryptography_InvalidHandle, nameof(safeCertContext)));
+
+            int dwErrorCode = ERROR_SUCCESS;
+
+            // Initialize view structure.
+            Interop.CryptUI.CRYPTUI_VIEWCERTIFICATE_STRUCTW ViewInfo = new Interop.CryptUI.CRYPTUI_VIEWCERTIFICATE_STRUCTW();
+            ViewInfo.dwSize = (uint)Marshal.SizeOf(ViewInfo);
+            ViewInfo.hwndParent = hwndParent;
+            ViewInfo.dwFlags = 0;
+            ViewInfo.szTitle = null;
+            ViewInfo.pCertContext = safeCertContext.DangerousGetHandle();
+            ViewInfo.rgszPurposes = IntPtr.Zero;
+            ViewInfo.cPurposes = 0;
+            ViewInfo.pCryptProviderData = IntPtr.Zero;
+            ViewInfo.fpCryptProviderDataTrustedUsage = false;
+            ViewInfo.idxSigner = 0;
+            ViewInfo.idxCert = 0;
+            ViewInfo.fCounterSigner = false;
+            ViewInfo.idxCounterSigner = 0;
+            ViewInfo.cStores = 0;
+            ViewInfo.rghStores = IntPtr.Zero;
+            ViewInfo.cPropSheetPages = 0;
+            ViewInfo.rgPropSheetPages = IntPtr.Zero;
+            ViewInfo.nStartPage = 0;
+
+            // View the certificate
+            if (!Interop.CryptUI.CryptUIDlgViewCertificateW(ViewInfo, IntPtr.Zero))
+                dwErrorCode = Marshal.GetLastWin32Error();
+
+            // CryptUIDlgViewCertificateW returns ERROR_CANCELLED if the user closes
+            // the window through the x button or by pressing CANCEL, so ignore this error code
+            if (dwErrorCode != ERROR_SUCCESS && dwErrorCode != ERROR_CANCELLED)
+                throw new CryptographicException(dwErrorCode);
+
+            safeCertContext.Dispose();
```

Comment:
This is a weird ownership semantic.  Being in a using statement from the caller makes a lot more sense.